lib/Sema/CSStep.cpp

```diff
@@ -354,6 +354,22 @@ StepResult ComponentStep::take(bool prevFailed) {
     return finalize(/*isSuccess=*/false);
   }
 
+  // If we don't have any disjunction or type variable choices left, we're done
+  // solving. Make sure we don't have any unsolved constraints left over, using
+  // report_fatal_error to make sure we trap in release builds instead of
+  // potentially miscompiling.
+  if (!CS.ActiveConstraints.empty()) {
+    llvm::report_fatal_error("Active constraints left over?");
```

Comment:
I'm fine with trapping here, but can we dump the constraint system before we do so? We might as well get something more meaningful out of the bug reports if this happens in the field.

Reply:
Good idea! Thanks for the review!